### DISCUSSION TITLE
Support for CiTO (Citation Typing Ontology)

### DISCUSSION THREAD

**Discussion by RichardJActon** [quarto-dev/quarto-cli#4176] [Feature Requests]
I think support for CiTO would be a really interesting addition to the bibliographic features in Quarto. It lets you document the way in which you are citing a source so that you can see for example that an article has been cited many times because people disagree with it's conclusions instead rather than because people agree with it.

It'd be cool to be able to have autocomplete prompts for the citation ontology terms when you are adding a citation in the visual editor in RStudio and have the terms appear in front of the citations in the bibliography & hover-over tooltips

I'm not sure how you'd go about implementing it though.

![image](https://user-images.githubusercontent.com/6893043/216042801-e5c50792-2fb2-470b-9fb3-363c7f490bb3.png)

- website: http://www.sparontologies.net/ontologies/cito
- Original reference: https://doi.org/10.1016/j.websem.2012.08.001
- Recent example of adoption:
  - https://doi.org/10.1186/s13321-020-00448-1
  - https://www.biomedcentral.com/collections/cito
- Docs: https://sparontologies.github.io/cito/current/cito.html

**Comment by cscheid** [comment]
We've talked about using shortcodes for this sort of stuff (I'd love to do it too. I'm a former academic from a field that does not have a rich citation culture like, for example, law does, and I agree that we need more of this)

With that said, a feature like this is at least 12-18 months away in terms of dependencies on other features that are on our plans. So, nothing concrete to report in the short-to-medium term.

**Reply by nichtich** [reply]
This should better be discussed in [pandoc](https://github.com/jgm/pandoc) anyway to agree on a CiTO in Markdown syntax not limited to quarto. The syntax variant [proposed here](https://jcheminform.github.io/jcheminform-author-guidelines/cito-guidelines/markdown.html) by @egonw to extend [pandoc citation syntax](https://pandoc.org/MANUAL.html#extension-citations) is just *one* possible way how it could be implemented and it needs to be aligned with [other extensions](https://talk.commonmark.org/t/cross-references-and-citations/4013).

**Reply by RichardJActon** [reply]
Egon also mentioned CiTO support pandoc in this pre-print about adding CiTO to BioHackrXiv pre-prints https://biohackrxiv.org/6rjvc I mentioned this thread to him on mastodon earlier.

**Reply by nichtich** [reply]
Egon writes `Support in pandoc of CiTO annotations dates back to 2017 [@usesMethodIn:Krewinkel2017]` but this is misleading. I'd say use of CiTO annotates *with* pandoc has been *illustrated* back in 2017. This is no feature of pandoc but a creative custom hack on top of pandoc citation keys - standalone pandoc would result in errors when using this syntax extension! The CiTO type used by Egon is more clear about it: It's `usesMethodIn:Krewinkel2017` not `usesMethodIn:pandoc`. [Krewinkel writes](https://peerj.com/articles/cs-112/):

> Our method is based on CiTO [...] The type of a citations can be added to a Markdown citation using @CITO_PROPERTY:KEY, where CITO_PROPERTY is a supported CiTO property, and KEY is the usual citation key. Our tool extracts that information and includes it in the generated linked data output.

Pandoc is being augmented by many tools, plugins and convention; quarto also adds some of these non-standard syntax-extensions, e.g. [callout blocks](https://quarto.org/docs/authoring/callouts.html). I welcome CiTO support in quarto but standard support in pandoc would be much better. So the right way would be to start a discussion at https://github.com/jgm/pandoc/issues and at https://talk.commonmark.org/t/cross-references-and-citations/4013 to get feedback from pandoc developers and users.

**Reply by egonw** [reply]
> Egon writes `Support in pandoc of CiTO annotations dates back to 2017 [@usesMethodIn:Krewinkel2017]` but this is misleading.

That was not intentional. I totally agree with native Pandoc support would be ideal.

We plan to make a revision of the preprint this weekend. I will rephrase the sentence.

**Reply by RichardJActon** [reply]
Thanks for the clarifications,

Just thinking out-loud here for a moment about what to put in a potential issue on the pandoc github page.

looking again at the pandoc citation syntax:

https://pandoc.org/MANUAL.html#citations-in-note-styles

It seems that there is already a way in which citations are qualified with additional information not in the bibliographic source data but at the point of citation in the text which is then included in the generated bibliography. Specifically the example in the docs includes a page number. I wonder if whatever mechanism which provides for this could be generalized - I've not dug through the code to see how exactly this is handled.

`[@foo, p.22]` or inline `@foo [p.22]`

With the qualification that I'm quite ignorant of native pandoc syntax I wonder if something along the lines of this:

`[@foo, p.22, [cito:usesMethodIn]]` or inline `@foo [p.22, [cito:usesMethodIn]]`

might work.

With multiple references and CiTO terms: 

`[@foo, p.22, [cito:usesMethodIn]; @bar, [cito:citesAsAuthority] [cito:agreesWith]]`

**Reply by nichtich** [reply]
A citation in pandoc [is internally stored](https://hackage.haskell.org/package/pandoc-types-1.23/docs/Text-Pandoc-Definition.html#t:Citation) with optional prefix and suffix. The locator such as "p. 22" is just an optional part of the suffix, extracted from it by pandoc-citeproc before the citation data is passed [as CitationItem](https://hackage.haskell.org/package/citeproc-0.8.1/docs/Citeproc-Types.html#t:CitationItem) to formatting. This is best described at

https://pandoc.org/MANUAL.html#extension-citations

This is a valid citation in pandoc markdown with prefix "introduced by", citation key "foo", locator "p. 22" and suffix "now standard":

    [introduced by @foo, p. 22 now standard]

So special keywords such as these could be used as well without any additional syntax:

    [cito:usesMethodIn @foo, p. 22 cito:citesAsAuthority]

Sure the specific syntax can be discussed, e.g.

    [(usesMethodIn) @foo, p. 22 (citesAsAuthority)]

To *fully* integrate CiTO in the citation ecosystem, I think it should be made part of [CSL](https://citationstyles.org/) so citation styles can choose whether and how to show references based on CiTO type. By now, suffix, prefix or maybe locator seem better places to pass through the additional information instead of adding it to the citating key.

**Reply by egonw** [reply]
> To _fully_ integrate CiTO in the citation ecosystem, I think it should be made part of [CSL](https://citationstyles.org/) 

Yes, agreed. But we have a bit of a chicken-and-egg situation here. It won't all happen in one step, and if no one makes a step, not one else will either. I will be excited about any current form of implementation. We have to start somewhere.

**Reply by nichtich** [reply]
Yes thanks for the initiative and for building a working prototype! Extending CSL is no easy task, so the approach to first show it should and can be done, then encourage adoption and discuss with community of interest (Pandoc, Quarto...) and finally get it implemented as standard feature seems promising!

**Reply by RichardJActon** [reply]
Thanks @nichtich that's very useful to know. So with the current pre/suffix features in pandoc we could adopt the convention of something like this `[@foo [cito:citesAsAuthority] [cito:agreesWith]; @bar]` based on the syntax adopted by Journal of Cheminformatics to represent the CiTO terms in their bibliographies (e.g. https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00448-1#Bib1) in their trial with @egonw. But different from their syntax for adding a citation in the body of the text: https://jcheminform.github.io/jcheminform-author-guidelines/cito-guidelines/markdown.html and placing the CiTO terms in the .bib notes. Though as you say this syntax might want some further discussion.

One of the complications I can see arising from having the CiTO terms in the .bib file is I might want to cite the same source from a .bib file with different CiTO terms in different documents with the same set underlying bibliography managed by Zotero or similar. Putting them in the suffix in pandoc would at least not cause this problem...  though I'm not sure what happens if you cite the same thing in more than one place in the same document with different pre/suffixes I'll have to check that. 

A plugin/extension that does `[@foo [cito:<tab auto-completion for cito terms>]` in VScode &/or RStudio visual mode could help out with adoption of this approach if we wanted to adopt something like this as an semi-formal convention for CiTO in pandoc.

Though yes adoption in CSL would be better for the wider citation ecosystem beyond pandoc. 

I think I'll try this approach out in the ebook I'm writing in Quarto atm and see how it turns out.

**Reply by RichardJActon** [reply]
Interesting I tried this out in [my book](https://hdbi.gitlab.io/data-management/hdbi-data-resource/) and it is not doing quite what I thought it would. The suffix is in-line at the point of the citation and not in the bibliography. My interpretation of the pandoc docs was that this wouldn't be visible in-line and would be in the bibliography.

**Reply by egonw** [reply]
> Egon writes `Support in pandoc of CiTO annotations dates back to 2017 [@usesMethodIn:Krewinkel2017]` but this is misleading.

@nichtich, I reworded this now to: `Support of CiTO annotations in the pandoc platform via extensions dates back to 2017 [@usesMethodIn:Krewinkel2017].`

**Comment by egonw** [comment]
A summary of the Journal of Cheminformatics CiTO Pilot has been published now: 
 https://jcheminf.biomedcentral.com/articles/10.1186/s13321-023-00683-2

The Pilot has ended and the matching Editorial is here: https://jcheminf.biomedcentral.com/articles/10.1186/s13321-023-00684-1

**Reply by RichardJActon** [reply]
Thanks for the update @egonw, shame it was discontinued though I can understand why from your editorial. Seems like improving the tooling so it's not a nightmare to actually use would be a major help towards cracking that chicken & egg problem.

**Reply by egonw** [reply]
Actually, the Pilot is over. The journal will continue to accept articles with CiTO annotation (and do their best). Btw, the decision was not mine (I am not an editor anymore).

**Comment by cderv** [comment]
As I don't think it was share here (or couldn't find it) but seems directly related to CiTo support with Pandoc, there is Lua filter available for that, that is organized in a way it can be used as a Quarto extension https://github.com/pandoc-ext/cito

Disclaimer: did not try it yet, and don't know much about it. I just wanted to share in this discussion that it exists
